src/intersections/lineray.jl

```diff
@@ -0,0 +1,33 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+#=
+The intersection type can be one of four types:
+
+1. intersect at one inner point (CrossingLineRay -> Point)
+2. intersect at origin of ray (TouchingLineRay -> Point)
+3. overlap of line and ray (OverlappingLineRay -> Ray)
+4. do not overlap nor intersect (NoIntersection)
+=#
+function intersection(f, l1::Line{N,T}, r1::Ray{N,T}) where {N,T}
```

Comment:
We also need to rename the order of arguments here to match the new naming convention.

Reply:
done ;)